Calendar/BookingHelpers/BookingMBLogic.ps1

```diff
@@ -0,0 +1,55 @@
+﻿# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+function RunMBTests {
+    [ref]$ErrorMessage = $null
+    [ref]$WriteMessageAlways = $true
+    $TestResult = $true
+
+    Write-DashLineBoxColor "Running Mailbox Tests" -Color Blue
+
+    $ErrorMessage = ""
+    $TestResult = CheckIdentityIsBookingsMB -errorMessage $ErrorMessage
+    WriteTestResult "Mailbox is Scheduling type " -success $TestResult -errorMessage $ErrorMessage
+
+    $TestResult = CheckIfMBIsHiddenInGAL
+    WriteTestResult "Is MB Hidden in GAL" -success $TestResult -errorMessage $ErrorMessage
+
+    $TestResult = CheckBookingsMBEmailAddresses -errorMessage $ErrorMessage
+    WriteTestResult "Check MB Email Addresses" -success $TestResult -errorMessage $ErrorMessage -writeMessageAlways $WriteMessageAlways
+    WriteBookingsMBEmailAddresses
+}
+
+function CheckIdentityIsBookingsMB {
+    param([ref]$ErrorMessage)
+    Write-Verbose "Checking if mailbox $Identity is Bookings mailbox"
+    if ($script:BookingMBData.RecipientTypeDetails -ne "SchedulingMailbox") {
+        $ErrorMessage.Value ="Mailbox $Identity is not a Bookings mailbox " + $script:BookingMBData.RecipientTypeDetails
+        return $false
+    }
+    return $true
+}
+function CheckIfMBIsHiddenInGAL {
+    Write-Verbose "Checking if a MB is Hidden in the GAL"
+    if ($script:BookingMBData.HiddenFromAddressListsEnabled -eq $true) {
+        $ErrorMessage.Value = "Mailbox $Identity is Hidden in the GAL."
```

Comment:
Pass in or this should be a `$Script:` scoped variable.

Reply:
Done